### DISCUSSION TITLE
Error when trying to set x-data with array

### DISCUSSION THREAD

**Discussion by FredMar** [alpinejs/alpine#3443] [1. Help] [answered]
Hi,

I can't figure out what the problem is here:

![image](https://user-images.githubusercontent.com/16269671/222500391-7c46afc4-a367-4bf8-b4ea-936ee4572817.png)

This does not work:
![image](https://user-images.githubusercontent.com/16269671/222501276-ff979879-cfbe-4aca-b9fd-e6429f427473.png)

But this does:
![image](https://user-images.githubusercontent.com/16269671/222501406-d5da2ab8-9969-4526-8eb8-dd47682a1799.png)

What I'm trying to do is this:
![image](https://user-images.githubusercontent.com/16269671/222501619-6ef66edf-115d-4b60-a000-3ee6a7882f4a.png)

Hope someone can help me.

Thanks :)

**Comment by ekwoka** [comment, marked as answer]
You reference a key that doesn't exist on the item.

**Reply by FredMar** [reply]
Thank you for taking the time to answer :)

Could you maybe be a little more specific: 

![image](https://user-images.githubusercontent.com/16269671/222672623-f69a28c9-bc4c-47ab-aa7b-6adfea376258.png)

Thanks :)

**Reply by ekwoka** [reply]
your `x-for` has `:key="el.id"` but your data does not have an `id` on it, so it's always undefined.

**Reply by FredMar** [reply]
Omg. I did not see that one - thank you so much!!

**Reply by ekwoka** [reply]
yeah, the error kind of sucks. Might be worth clarifying it with a console log and defaulting to like...no key.
